### ISSUE TITLE
Is Service Worker cache supported in WebView2?

### ISSUE THREAD

**Issue by Mintyboi** [MicrosoftEdge/WebView2Feedback#1757]
I am trying out the C++ sdk for WebView2 and am using the Microsoft.Web.WebView2 v1.0.1018-prerelease package.
My goal is to be able to start a Service Worker and cache my resources in the Service Worker Cache.

I read an earlier [thread](https://github.com/MicrosoftEdge/WebView2Feedback/issues/362#issuecomment-670076389) that mentioned that caching is probably working in WebView2. However, I seem to have some trouble getting it to work.

This is my sw code which works well in the browser:
```javascript
let CACHENAME = "TEST"

self.addEventListener('fetch', (event) => {
    console.log("Service Worker Fetching: ", event.request.url);
    event.respondWith(
        caches.match(event.request).then((cachedResponse) => {
            if (cachedResponse) {
                console.log("Found cache for ", event.request.url);
                return cachedResponse;
            }
            return fetch(event.request).then((response) => {
                const responseToCache = response.clone();
                return caches.open(CACHENAME).then((cache) => {
                    console.log("Caching into SW now: ", event.request.url);
                    cache.put(event.request, responseToCache);
                    return response;
                })
                .catch(err => {
                    console.error("Cache open error: ", err);
                    return undefined;
                });
            })
            .catch(err => {
                console.error("Fetch error: ", err);
                return undefined;
            });
        })
    );
});

self.addEventListener('install', (event) => {
    console.log("Service Worker Installing");
    event.waitUntil(
        caches.open(CACHENAME).then(function (cache) {
            return cache.addAll(
                ['/edge.png']
            );
        })
        .catch(err => {
            // DOMException: Entry already exists.
            console.error("Cache error: ", err);
            return undefined;
        })
    );
});

self.addEventListener('activate', (event) => {
     console.log("Service Worker Activating");
});
```

Issue 1:
In install, I attempt to fetch and store an image, however I got an `DOMException: Entry already exists.` exception

Issue 2:
In fetch, I am able to fetch the resource, open the cache and call put() without any errors. However the cache storage in the Dev Tool is empty, and on reload, the resource cannot be found as well.

Is Service Worker Cache officially supported? Or is there something in my project that I'm missing?

**Comment by champnic** [comment]
Hey @Mintyboi - thanks for the tag! Sorry I didn't get to this sooner. I'm checking with the team on this one and will get back to you soon.

**Comment by champnic** [comment]
@Mintyboi I've confirmed that service worker caching should be working.

Are you able to share a small sample app that reproduces this issue?

**Comment by Mintyboi** [comment]
@champnic Thanks for checking on this and your prompt reply.

I was using and extending the WebView2 [sample application](https://github.com/MicrosoftEdge/WebView2Samples). However when I tried to pull the project out to create a reproducer, the SW cache was magically working for me this time.
Although it continues not to work when I'm in the sample application for some reason, but at least now I know that the SW cache is working well. Sorry for any trouble caused.

**Comment by champnic** [comment]
No problem! Glad you were able to verify it can work. If you do find a specific cause for it not working and think it's due to WebView2 please don't hesitate to reach out again. Thanks!

**Comment by Mintyboi** [comment]
@champnic I found something new on this issue. I believe the `DOMException: Entry already exists.` error happens when the path to the Service Worker cache is too long.

I find that when I have 117 or more characters before the EBWebView folder, I get this error.
Here is the path on my local computer: `C:\TEMP\WebView2SWcacheissue\a7c66f9ab547a4e60dc59618116a46612b7b05d4a7c66f9ab547a4e0s\WebView2APISample.exe.WebView2`

I've attached a sample project that can reproduce this issue when placed at the correct place.
Reproduce steps:
1. Launch WebView2APISample.exe
2. Right click and click on Inspect to launch the dev console
3. Go to Scenario > Virtual Host Mapping with Service Worker
4. Go back to the dev console and look at the console tab

[WebView2SWcacheissue.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/7451935/WebView2SWcacheissue.zip)

Is this an issue with WebView2? Or will you attribute it to chromium?
